src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -91,8 +96,77 @@ public void failed(Throwable exc, Void v) {
         }
     }
 
-    /** Returns a file object for the given cache key. */
-    File getFileForKey(String key) {
-        return new File(mRootDirectorySupplier.get(), DiskBasedCacheUtility.getFilenameForKey(key));
+    /** Puts the cache entry with a specified key into the cache. */
+    @Override
+    public void put(final String key, Cache.Entry entry, final OnPutCompleteCallback callback) {
+        // If adding this entry would trigger a prune, but pruning would cause the new entry to be
+        // deleted, then skip writing the entry in the first place, as this is just churn.
+        // Note that we don't include the cache header overhead in this calculation for simplicity,
+        // so putting entries which are just below the threshold may still cause this churn.
+        if (mTotalSize + entry.data.length > mMaxCacheSizeInBytes
+                && entry.data.length
+                        > mMaxCacheSizeInBytes * DiskBasedCacheUtility.HYSTERESIS_FACTOR) {
```

Comment:
Should these `>` signs be replaced with `>=`? Looking at `pruneIfNeeded`, my understanding is that if the new size is bigger or equal to the max size, the cache would be pruned.

Reply:
I didn't write these checks, so I'm not totally sure why this is, but it does look like the cache is pruned if the total size if the same as the max size. Would be want to change the check, or change in pruneIfNeeded?

Reply:
To keep it functionally the same, I'd update these. Note that if you have a method to check those conditions (see the other comment), we wouldn't need to update them because the shared method would take care of keeping the check consistent for us.